aws/resource_aws_kinesisanalyticsv2_application.go

```diff
@@ -0,0 +1,2640 @@
+		output.DestinationSchema = expandKinesisAnalyticsV2DestinationSchema(vDestinationSchema)
+	}
+
+	if vKinesisFirehoseOutput, ok := mOutput["kinesis_firehose_output"].([]interface{}); ok && len(vKinesisFirehoseOutput) > 0 && vKinesisFirehoseOutput[0] != nil {
+		kinesisFirehoseOutput := &kinesisanalyticsv2.KinesisFirehoseOutput{}
+
+		mKinesisFirehoseOutput := vKinesisFirehoseOutput[0].(map[string]interface{})
+
+		if vResourceArn, ok := mKinesisFirehoseOutput["resource_arn"].(string); ok && vResourceArn != "" {
+			kinesisFirehoseOutput.ResourceARN = aws.String(vResourceArn)
+		}
+
+		output.KinesisFirehoseOutput = kinesisFirehoseOutput
+	}
+
+	if vKinesisStreamsOutput, ok := mOutput["kinesis_streams_output"].([]interface{}); ok && len(vKinesisStreamsOutput) > 0 && vKinesisStreamsOutput[0] != nil {
+		kinesisStreamsOutput := &kinesisanalyticsv2.KinesisStreamsOutput{}
+
+		mKinesisStreamsOutput := vKinesisStreamsOutput[0].(map[string]interface{})
+
+		if vResourceArn, ok := mKinesisStreamsOutput["resource_arn"].(string); ok && vResourceArn != "" {
+			kinesisStreamsOutput.ResourceARN = aws.String(vResourceArn)
+		}
+
+		output.KinesisStreamsOutput = kinesisStreamsOutput
+	}
+
+	if vLambdaOutput, ok := mOutput["lambda_output"].([]interface{}); ok && len(vLambdaOutput) > 0 && vLambdaOutput[0] != nil {
+		lambdaOutput := &kinesisanalyticsv2.LambdaOutput{}
+
+		mLambdaOutput := vLambdaOutput[0].(map[string]interface{})
+
+		if vResourceArn, ok := mLambdaOutput["resource_arn"].(string); ok && vResourceArn != "" {
+			lambdaOutput.ResourceARN = aws.String(vResourceArn)
+		}
+
+		output.LambdaOutput = lambdaOutput
+	}
+
+	if vName, ok := mOutput["name"].(string); ok && vName != "" {
+		output.Name = aws.String(vName)
+	}
+
+	return output
+}
+
+func expandKinesisAnalyticsV2Outputs(vOutputs []interface{}) []*kinesisanalyticsv2.Output {
+	outputs := []*kinesisanalyticsv2.Output{}
+
+	for _, vOutput := range vOutputs {
+		output := expandKinesisAnalyticsV2Output(vOutput)
+
+		if output != nil {
+			outputs = append(outputs, expandKinesisAnalyticsV2Output(vOutput))
+		}
+	}
+
+	return outputs
+}
+
+func expandKinesisAnalyticsV2PropertyGroups(vPropertyGroups []interface{}) []*kinesisanalyticsv2.PropertyGroup {
+	propertyGroups := []*kinesisanalyticsv2.PropertyGroup{}
+
+	for _, vPropertyGroup := range vPropertyGroups {
+		propertyGroup := &kinesisanalyticsv2.PropertyGroup{}
+
+		mPropertyGroup := vPropertyGroup.(map[string]interface{})
+
+		if vPropertyGroupID, ok := mPropertyGroup["property_group_id"].(string); ok && vPropertyGroupID != "" {
+			propertyGroup.PropertyGroupId = aws.String(vPropertyGroupID)
+		} else {
+			// https://github.com/hashicorp/terraform-plugin-sdk/issues/588
+			continue
+		}
+
+		if vPropertyMap, ok := mPropertyGroup["property_map"].(map[string]interface{}); ok && len(vPropertyMap) > 0 {
+			propertyGroup.PropertyMap = stringMapToPointers(vPropertyMap)
```

Comment:
stringMapToPointers  --> aws.StringMap

Reply:
I don't think [`aws.StringMap`](https://docs.aws.amazon.com/sdk-for-go/api/aws/#StringMap) will work here as `vPropertyMap` is `map[string]interface{}`, not `map[string]string`.